src/Client/ClickHouseAsyncClient.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\ClickHouseClient\Client;
+
+use GuzzleHttp\Promise\PromiseInterface;
+use SimPod\ClickHouseClient\Format\Format;
+use SimPod\ClickHouseClient\Output\Output;
+
+interface ClickHouseAsyncClient
```

Comment:
tohle je první věc co mě zarazila - asi bych to nerozděloval ... fakt ti stojí za to předávat někdy dvě třídy, jen protože jedna metoda je v jiném interface? reálně ti rozdělení nic nepřinese imo, jen právě práci navíc

Reply:
jj. asi tu bude i vic metod a dependency na promises pak muze bejt optional

+ jsou ty tridy pak lighter

Reply:
okey no, mě to api přijde debilní takhle no, ale jak chceš